go.mod

```diff
@@ -1,84 +1,91 @@
+replace k8s.io/cri-api => k8s.io/cri-api v0.20.12-rc.0
 
-replace k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.18.5
+replace k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.20.10
 
-replace k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.18.5
+replace k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.20.10
 
-replace k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.18.5
+replace k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.20.10
 
-replace k8s.io/kube-proxy => k8s.io/kube-proxy v0.18.5
+replace k8s.io/kube-proxy => k8s.io/kube-proxy v0.20.10
 
-replace k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.18.5
+replace k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.20.10
 
-replace k8s.io/kubectl => k8s.io/kubectl v0.18.5
+replace k8s.io/kubectl => k8s.io/kubectl v0.20.10
 
-replace k8s.io/kubelet => k8s.io/kubelet v0.18.5
+replace k8s.io/kubelet => k8s.io/kubelet v0.20.10
 
-replace k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.18.5
+replace k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.20.10
 
-replace k8s.io/metrics => k8s.io/metrics v0.18.5
+replace k8s.io/metrics => k8s.io/metrics v0.20.10
 
-replace k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.18.5
+replace k8s.io/mount-utils => k8s.io/mount-utils v0.20.11-rc.0
 
-replace k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.18.5
+replace k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.20.10
 
-replace k8s.io/sample-controller => k8s.io/sample-controller v0.18.5
+replace k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.20.10
 
-replace sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.6.0
+replace k8s.io/sample-controller => k8s.io/sample-controller v0.20.10
+
+require (
+	github.com/agiledragon/gomonkey v2.0.2+incompatible
+	github.com/brahma-adshonor/gohook v1.1.9
+	github.com/container-storage-interface/spec v1.2.0
+	github.com/docker/go-units v0.4.0
+	github.com/go-logr/logr v0.4.0
+	github.com/go-openapi/spec v0.19.3
+	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
+	github.com/golang/mock v1.4.1
+	github.com/golang/protobuf v1.5.2 // indirect
+	github.com/kubernetes-csi/csi-lib-utils v0.9.1 // indirect
+	github.com/kubernetes-csi/drivers v1.0.2
+	github.com/onsi/ginkgo v1.14.1
+	github.com/onsi/gomega v1.10.2
+	github.com/pkg/errors v0.9.1
+	github.com/smartystreets/goconvey v1.6.4
+	github.com/spf13/cobra v1.1.1
+	github.com/stretchr/testify v1.7.0 // indirect
+	go.uber.org/zap v1.15.0
+	golang.org/x/net v0.0.0-20210520170846-37e1c6afe023
+	google.golang.org/grpc v1.38.0
+	gopkg.in/yaml.v2 v2.3.0
+	k8s.io/api v0.22.0
+	k8s.io/apimachinery v0.20.10
+	k8s.io/client-go v0.22.0
+	k8s.io/component-base v0.22.0 // indirect
+	k8s.io/component-helpers v0.20.10
+	k8s.io/klog v1.0.0
+	k8s.io/klog/v2 v2.9.0 // indirect
+	k8s.io/kube-openapi v0.0.0-20201113171705-d219536bb9fd
+	k8s.io/kubernetes v1.20.10
```

Comment:
Can we not use Kubernetes pkg directly ?

Reply:
#958

Reply:
I'm using `bash k8s-mod.sh v1.20.10` to upgrade K8s version.  This script is from K8s and using the exact same version in K8s repo.

Reply:
Now it's fixed. See the latest commit.

Reply:
awesome !
Because if Fluid uses the kubernetes package directly, a project that references Fluid must also maintain replace in go.mod, 
This is not an official recommended approach, https://github.com/kubernetes/kubernetes/issues/90358#issuecomment-617859364

Reply:
I was using same useful library, and I am afraid it's not Inevitable.

I was using the solution in https://github.com/kubernetes/kubernetes/issues/79384. It works as a charm.

Reply:
OK, I got it